common/compositor/gl/glprogram.cpp

```diff
@@ -298,36 +308,39 @@ GLProgram::GLProgram()
       alpha_loc_(0),
       premult_loc_(0),
       tex_matrix_loc_(0),
-      initialized_(false) {
+      previous_layer_size_(0) {
 }
 
 GLProgram::~GLProgram() {
   if (program_ != 0)
     glDeleteProgram(program_);
 }
 
-bool GLProgram::Init(unsigned texture_count) {
+bool GLProgram::Init() {
   std::ostringstream shader_log;
-  program_ = GenerateProgram(texture_count, &shader_log);
+  if (!program_)
+    program_ = GenerateProgram(&shader_log);
   if (!program_) {
     ETRACE("%s", shader_log.str().c_str());
     return false;
   }
-
   return true;
 }
 
 void GLProgram::UseProgram(const RenderState &state, GLuint viewport_width,
                            GLuint viewport_height) {
   glUseProgram(program_);
   unsigned size = state.layer_state_.size();
-  if (!initialized_) {
+  if (size != previous_layer_size_) {
```

Comment:
Why should we need this check?
if program is reused, , texture size is changing

Reply:
Please use !initialized, texture uniform use MAX_LAYER_COUNT

Reply:
Should avoid duplidate initliazation